### PR TITLE
Added simple_load module

### PR DESCRIPTION
**Pull request by jonathanlunt**
- Added `simple_load` module to complement `simple_save`. This is useful for testing your generated model artifact in python as well as reloading a previously saved model for batch prediction at a later date.
- Updated `saved_model` documentation to include references to both `simple_save` and `simple_load`. Previously this was not mentioned at all.

### CODE REVIEW

**Review by drpngx — CHANGES_REQUESTED** [review]
Overall, seems reasonable to me. @sukritiramesh can you confirm?

File: tensorflow/python/saved_model/README.md [hunk 1]
```diff
@@ -76,7 +76,56 @@ saved_model.pb
...
+
+The highest level API for exporting the model components described above
+is using the [simple_save](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/simple_save.py)
+module. For most use-cases this functionality is sufficient to export
+a model which is compatible with TensorFlow serving. A corresponding [simple_load](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/simple_load.py)
+module is implemented in python which enables SavedModels to be reloaded
+within a python TensorFlow session.
+
+#### Simple Save Usage
+~~~python
```
[the remark this answers is not in the document]
Comment (jonathanlunt) [reply tensorflow/python/saved_model/README.md]: I was trying to be consistent with the existing code-blocks by using `~~~`. I have updated the entire `saved_model/README.md` file to use backticks

File: tensorflow/python/saved_model/README.md [hunk 2]
```diff
@@ -76,7 +76,56 @@ saved_model.pb
 ## APIs
 The APIs for building and loading a SavedModel are described in this section.
 
+### Simple
+
+The highest level API for exporting the model components described above
+is using the [simple_save](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/simple_save.py)
```
Comment (drpngx) [inline tensorflow/python/saved_model/README.md]: Use `@{tf.saved_model.simple_save}` and load everywhere. For top-level info, use `@{$guide/saved_model}`.

**Comment by drpngx** [comment]
Thanks! Please also run the test suite and update the API goldens.

**Comment by jonathanlunt** [comment]
- Added golden API docs. It could use some documentation to mention that you need to configure your build with python 2 to test/update the golden API.
- Ran tests with `bazel test --config=opt //tensorflow/python/saved_model/...` and got successful output:
```
INFO: Build completed successfully, 1 total action
//tensorflow/python/saved_model:loader_test                     (cached) PASSED in 2.2s
//tensorflow/python/saved_model:saved_model_test                (cached) PASSED in 3.9s
//tensorflow/python/saved_model:signature_def_utils_test        (cached) PASSED in 1.4s
//tensorflow/python/saved_model:simple_load_test                (cached) PASSED in 2.0s
//tensorflow/python/saved_model:simple_save_test                (cached) PASSED in 1.6s
//tensorflow/python/saved_model:utils_test                      (cached) PASSED in 1.4s
```

**Comment by asimshankar** [comment]
(API is addition is good)

**Comment by karmel** [comment]
@k-w-w , can you comment on whether this is still necessary given recent changes you have made?

**Comment by k-w-w** [comment]
I think this would be a good addition as the `simple_save` complement. The `SavedModelLoader` still requires the user to learn about the inner workings of a SavedModel (Metagraph tags, and SignatureDef keys), and `SavedModelEstimator` loads the SavedModel with the estimator framework (which the user may not be familiar with).

`simple_save` and `simple_load` covers the use case for building a graph and loading it back in using Python with minimal overhead.

File: tensorflow/python/saved_model/simple_load.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+"""SavedModel simple load functionality."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import tag_constants
+from tensorflow.python.util.tf_export import tf_export
+from tensorflow.python.saved_model import loader
```
Comment (k-w-w) [inline tensorflow/python/saved_model/simple_load.py]: lint: import order (should be alphabetical, so loader should be first)

File: tensorflow/python/saved_model/saved_model.py [hunk 4]
```diff
@@ -29,8 +29,6 @@
 from tensorflow.python.saved_model import signature_def_utils
 from tensorflow.python.saved_model import tag_constants
 from tensorflow.python.saved_model import utils
+from tensorflow.python.saved_model.simple_save import simple_save
```
Comment (k-w-w) [inline tensorflow/python/saved_model/saved_model.py]: lint import order here as well

**Review by k-w-w — DISMISSED** [review]
Look good to me, except the import order lint.

**Comment by karmel** [comment]
Looks like you'll need to rerun the API goldens as well, @jonathanlunt . After that + the lint fixes above, we can pull and merge-- many thanks

File: tensorflow/python/saved_model/simple_load.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+@tf_export('saved_model.simple_load')
+def simple_load(session, export_dir):
+  """Load a SavedModel artifact into a python TensorFlow Session.
+
+  This function assumes that the export directory contains a model that has been
+  serialized using the @{tf.saved_model.simple_save} function.
+
+  More information about SavedModel and signatures can be found here in the
+  @{$guide/saved_model} documentation.
```
Comment (karmel) [inline tensorflow/python/saved_model/simple_load.py]: @MarkDaoust -- what's the new format for these magic doc decorators?

Comment (MarkDaoust) [reply tensorflow/python/saved_model/simple_load.py]: Code references can just be back-quoted `tf.saved_model.simple_save` and they'll automatically be linked when the site is generates.

We haven't sorted out doc references yet, so that @{$guide/saved_model} should be good for now.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by jonathanlunt** [comment]
I was going to make updates since I have time again now, but it seems like the SavedModel serialization is changing for the 2.0 release. I am going to close this because it will be irrelevant if the `simple_save` API goes away in future version.

Regarding serialization, I have an ask (@drpngx and others)... I hope core contributors are thinking very carefully about end-users when it comes to the updated serialization API. I am not looking forward to interfacing with the `saved_model.export` code in the future. It exposes additional internal objects/functions to the user and it seems primarily motivated by the Keras/Eager APIs. None of these things are useful to users like me.

**Comment by asimshankar** [comment]
CC @allenlavoie 
@jonathanlunt : Could you provide some specifics on what about the future serialization APIs is problematic? (Perhaps in a discussion on developers@tensorflow.org instead of the comments on this pull request :))
